### PR TITLE
Fix NNS compatibility

### PR DESCRIPTION
**Pull request by roman-khimov**

### CODE REVIEW

File: examples/nft-nd-nns/nns.go [hunk 1]
```diff
@@ -613,6 +613,9 @@ func checkIPv6(data string) bool {
 			nums[idx] = n
 		}
 	}
+	if l < 8 && !hasEmpty {
```
Comment (AnnaShaleva) [inline examples/nft-nd-nns/nns.go:647]: There's one more related reference PR.

File: examples/nft-nd-nns/nns.go [hunk 2]
```diff
@@ -353,6 +360,15 @@ func Resolve(name string, typ RecordType) string {
 	return resolve(ctx, name, typ, 2)
 }
 
+// GetAllRecords returns an Iterator with RecordState items for given name.
+func GetAllRecords(name string) iterator.Iterator {
```
Comment (AnnaShaleva) [inline examples/nft-nd-nns/nns.go:364]: Don't we want to add the corresponding API to RPC client?

File: examples/nft-nd-nns/nns.go (hunk 1 above)
Comment (roman-khimov) [reply examples/nft-nd-nns/nns.go:653]: Yep, it was added later.

File: examples/nft-nd-nns/nns.go (hunk 2 above)
Comment (roman-khimov) [reply examples/nft-nd-nns/nns.go:364]: Fixed.

File: examples/nft-nd-nns/nns.go [hunk 3]
```diff
@@ -649,10 +680,12 @@ func resolve(ctx storage.Context, name string, typ RecordType, redirect int) str
...
 	cname := ""
 	for iterator.Next(records) {
-		r := iterator.Value(records).([]string)
-		key := []byte(r[0])
-		value := r[1]
-		rTyp := key[len(key)-1]
+		r := iterator.Value(records).(struct {
+			key string
+			rs  RecordState
+		})
```
Comment (fyrchik) [inline examples/nft-nd-nns/nns.go:692]: Nice to see this working even though it is missing from compiler tests!

File: examples/nft-nd-nns/nns.yml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 name: "NameService"
 supportedstandards: ["NEP-11"]
 safemethods: ["balanceOf", "decimals", "symbol", "totalSupply", "tokensOf", "ownerOf",
-              "tokens", "properties", "roots", "getPrice", "isAvailable", "getRecord", "resolve"]
+              "tokens", "properties", "roots", "getPrice", "isAvailable", "getRecord",
+	      "resolve", "getAllRecords"]
```
Comment (fyrchik) [inline examples/nft-nd-nns/nns.yml:5]: White space issue (second line has a tab)

**Review by fyrchik — APPROVED** [review]

File: examples/nft-nd-nns/nns.yml (hunk 4 above)
Comment (roman-khimov) [reply examples/nft-nd-nns/nns.yml]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**